ENIGMAsystem/SHELL/Widget_Systems/General/tinyfiledialogs/dialogs.cpp

```diff
@@ -113,6 +113,38 @@ string tfd_add_escaping(string str)
   return result;
 }
 
+string detect_all_files_filter(string filter)
+{
+  if (tfd_DialogEngine() == tfd_OsaScript)
+  {
+    string str_filter = string_replace_all(filter, "*.", "");
+    size_t first_line_pos = str_filter.find("|");
+
+    if (first_line_pos != string::npos)
+    {
+      size_t first_star_pos = str_filter.find("*", first_line_pos + 1);
+            
+      if (first_star_pos != string::npos)
+      {
+        size_t second_line_pos = str_filter.find("|", first_line_pos + 1);
+                
+        if (second_line_pos != string::npos)
+        {
+          if (first_line_pos < first_star_pos && first_star_pos < second_line_pos)
+            filter = "";
```

Comment:
You're handling an error case by defaulting the format string to the empty string? What happens if you don't do this?

Reply:
We discussed it over Discord and he said it works just like GMS's port of the function.

>it works just like GameMaker Studio's port of get_open_filename on Mac. On Mac there are no filters, instead, allowed filetypes
That's why it only uses the first "filter"
That's how GMStudio works on Mac

Reply:
@RobertBColton it works just like GameMaker Studio's port of get_open_filename on Mac. On Mac there are no filters, instead, allowed filetypes. That's why it only uses the first "filter". That's how it work in GMStudio on Mac.